Idcs Role Mapper: NPE when user does not have any groups`

When mapping roles for a user without any IDCS groups, the "groups" field is not present in the IDCS response, causing NPE in Helidon:
```
io.helidon.security.SecurityException: Failure while executing asynchronous security
	at io.helidon.security.SecurityResponse.get(SecurityResponse.java:67) ~[helidon-security-1.0.0.jar:?]
	at io.helidon.security.SecurityClient.get(SecurityClient.java:46) ~[helidon-security-1.0.0.jar:?]
	at io.helidon.security.SecurityClientBuilder.buildAndGet(SecurityClientBuilder.java:61) ~[helidon-security-1.0.0.jar:?]
	at io.helidon.security.integration.jersey.SecurityFilterCommon.processAuthentication(SecurityFilterCommon.java:221) ~[helidon-security-integration-jersey-1.0.0.jar:?]
	at io.helidon.security.integration.jersey.SecurityFilterCommon.authenticate(SecurityFilterCommon.java:173) ~[helidon-security-integration-jersey-1.0.0.jar:?]
	at io.helidon.security.integration.jersey.SecurityFilter.processSecurity(SecurityFilter.java:159) ~[helidon-security-integration-jersey-1.0.0.jar:?]
	at io.helidon.security.integration.jersey.SecurityFilterCommon.doFilter(SecurityFilterCommon.java:146) ~[helidon-security-integration-jersey-1.0.0.jar:?]
	at io.helidon.security.integration.jersey.SecurityFilter.filter(SecurityFilter.java:145) ~[helidon-security-integration-jersey-1.0.0.jar:?]
	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:132) ~[jersey-server-2.26.jar:?]
	at org.glassfish.jersey.server.ContainerFilteringStage.apply(ContainerFilteringStage.java:68) ~[jersey-server-2.26.jar:?]
	at org.glassfish.jersey.process.internal.Stages.process(Stages.java:197) ~[jersey-common-2.27.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:269) [jersey-server-2.26.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:272) [jersey-common-2.27.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:268) [jersey-common-2.27.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:316) [jersey-common-2.27.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:298) [jersey-common-2.27.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:268) [jersey-common-2.27.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:289) [jersey-common-2.27.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:256) [jersey-server-2.26.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:703) [jersey-server-2.26.jar:?]
	at io.helidon.webserver.jersey.JerseySupport$JerseyHandler.lambda$accept$1(JerseySupport.java:210) [helidon-webserver-jersey-1.0.0.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2022) ~[?:?]
	at io.helidon.security.SecurityResponse.get(SecurityResponse.java:63) ~[helidon-security-1.0.0.jar:?]
	... 25 more
Caused by: java.lang.NullPointerException
	at io.helidon.security.providers.idcs.mapper.IdcsRoleMapperProvider.lambda$getGrantsFromServer$4(IdcsRoleMapperProvider.java:158) ~[helidon-security-providers-idcs-mapper-1.0.0.jar:?]
	at java.util.Optional.flatMap(Optional.java:294) ~[?:?]
	at io.helidon.security.providers.idcs.mapper.IdcsRoleMapperProvider.getGrantsFromServer(IdcsRoleMapperProvider.java:139) ~[helidon-security-providers-idcs-mapper-1.0.0.jar:?]
	at io.helidon.security.providers.idcs.mapper.IdcsRoleMapperProvider.lambda$enhance$2(IdcsRoleMapperProvider.java:115) ~[helidon-security-providers-idcs-mapper-1.0.0.jar:?]
	at io.helidon.security.providers.common.EvictableCacheImpl.lambda$doComputeValue$5(EvictableCacheImpl.java:143) ~[helidon-security-providers-common-1.0.0.jar:?]
	at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1908) ~[?:?]
	at io.helidon.security.providers.common.EvictableCacheImpl.doComputeValue(EvictableCacheImpl.java:133) ~[helidon-security-providers-common-1.0.0.jar:?]
	at io.helidon.security.providers.common.EvictableCacheImpl.computeValue(EvictableCacheImpl.java:98) ~[helidon-security-providers-common-1.0.0.jar:?]
	at io.helidon.security.providers.idcs.mapper.IdcsRoleMapperProvider.enhance(IdcsRoleMapperProvider.java:115) ~[helidon-security-providers-idcs-mapper-1.0.0.jar:?]
	at io.helidon.security.providers.idcs.mapper.IdcsRoleMapperProvider.lambda$map$0(IdcsRoleMapperProvider.java:107) ~[helidon-security-providers-idcs-mapper-1.0.0.jar:?]
	at java.util.Optional.map(Optional.java:265) ~[?:?]
	at io.helidon.security.providers.idcs.mapper.IdcsRoleMapperProvider.map(IdcsRoleMapperProvider.java:107) ~[helidon-security-providers-idcs-mapper-1.0.0.jar:?]
	at io.helidon.security.AuthenticationClientImpl.lambda$mapSubject$2(AuthenticationClientImpl.java:66) ~[helidon-security-1.0.0.jar:?]
	at java.util.Optional.map(Optional.java:265) ~[?:?]
	at io.helidon.security.AuthenticationClientImpl.mapSubject(AuthenticationClientImpl.java:66) ~[helidon-security-1.0.0.jar:?]
	at io.helidon.security.AuthenticationClientImpl.lambda$submit$1(AuthenticationClientImpl.java:51) ~[helidon-security-1.0.0.jar:?]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072) ~[?:?]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1705) ~[?:?]
```
## Environment Details
* Helidon Version: 1.0.0
* Helidon MP
* JDK version: 11.0.2
* OS: Linux
* Docker version (if applicable):

--- a/security/providers/idcs-mapper/src/main/java/io/helidon/security/providers/idcs/mapper/IdcsRoleMapperProvider.java
+++ b/security/providers/idcs-mapper/src/main/java/io/helidon/security/providers/idcs/mapper/IdcsRoleMapperProvider.java
@@ -153,6 +153,11 @@ public final class IdcsRoleMapperProvider implements SubjectMappingProvider {
                 JsonObject jsonObject = groupResponse.readEntity(JsonObject.class);
                 JsonArray groups = jsonObject.getJsonArray("groups");
 
+                if (null == groups) {
+                    LOGGER.finest(() -> "No groups found for user " + subject);
+                    return Optional.empty();
+                }
+
                 List<Role> result = new LinkedList<>();
 
                 for (int i = 0; i < groups.size(); i++) {

